providers/oidc.go

```diff
@@ -170,7 +164,16 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	newSession.AccessToken = token.AccessToken
 	newSession.RefreshToken = token.RefreshToken
 	newSession.CreatedAt = &created
-	newSession.ExpiresOn = &token.Expiry
+
+	var tokenExpiry time.Time
```

Comment:
This now makes the expiration set on refresh different than how we set it on redeem. If we are going to make this change, we should align it for both I think.

Also, we have multiple OIDC-compatible providers that will have IDTokens that I'm looking to globalize the OIDC logic for in this PR here: https://github.com/oauth2-proxy/oauth2-proxy/pull/936/files#diff-b4ef38e96ab5394c070aff6c070843820d904f7f06b36a81bd52384edd5214caR114

This logic potentially feels best as a helper method.

Reply:
I'll extract this logic into a helper method and will align both redeem and and refresh to be consistent.